### PR TITLE
Fixed ordering of quotation marks next to punctuation and symbols

### PR DESCRIPTION
**Pull request by matthew-ivre**
Test strings:
image

### CODE REVIEW

**Comment by matthew-ivre** [comment]
Extended tests - before (issues highlighted):
image

After:
image

File: Assets/RTLTMPro/Scripts/Runtime/LigatureFixer.cs [hunk 1]
```diff
@@ -130,11 +145,15 @@ public static void Fix(FastStringBuilder input, FastStringBuilder output, bool f
...
+                        // Check if the punctuation comes at the end of the string and follows a number
                         bool isAfterNumber = Char32Utils.IsNumber(previousCharacter, preserveNumbers, farsi);
                         bool isSpecialPunctuation = characterAtThisIndex is '.' or '،' or '؛' or '؟';
 
-                        if (isAfterNumber && isSpecialPunctuation)
+                        // Check if the punctuation comes at the end of the string and follows a quotation mark
+                        bool isAfterQuote = Char32Utils.IsQuote(previousCharacter);
+
+                        if (isAfterNumber && isSpecialPunctuation ||
+                            isAfterQuote && isSpecialPunctuation)
```
Comment (BenIVRE) [inline Assets/RTLTMPro/Scripts/Runtime/LigatureFixer.cs]: Double checking `isSpecialPunctuation` can be changed to `if (isSpecialPunctuation && (isAfterNumber || isAfterQuote))`

**Review by BenIVRE — CHANGES_REQUESTED** [review]
I think it would be good to include your test cases as actual unit tests. Let me know if you want a hand setting those up.

File: Assets/RTLTMPro/Scripts/Runtime/TextUtils.cs [hunk 2]
```diff
@@ -79,6 +79,12 @@ public static bool IsEnglishLetter(char ch)
                 || ch >= ArabicPresentationFormsBBlockLow && ch <= ArabicPresentationFormsBBlockHigh;
         }
 
+        // Checks if the character is a quote symbol. Either the generic " or guillemets
+        public static bool IsQuote(char input)
+        {
+            return input is '"' or '�' or '�';
```
Comment (ruben-ivre) [inline Assets/RTLTMPro/Scripts/Runtime/TextUtils.cs]: I would rather include those special chars as \uXXXX format rather than trust file encoding.

**Review by ruben-ivre — APPROVED** [review]

File: Assets/RTLTMPro/Scripts/Runtime/LigatureFixer.cs (hunk 1 above)
Comment (matthew-ivre) [reply Assets/RTLTMPro/Scripts/Runtime/LigatureFixer.cs]: Sure. Looks cleaner. Implemented in [9e13f89](https://github.com/engage-xr/RTLTMPro/pull/4/commits/9e13f896ba65ec6a957ef31d28e0ad77499cda2e)

File: Assets/RTLTMPro/Scripts/Runtime/TextUtils.cs (hunk 2 above)
Comment (matthew-ivre) [reply Assets/RTLTMPro/Scripts/Runtime/TextUtils.cs]: Makes sense. The original project declared character constants in 0xXXXX format, but I'm presuming there's no functional difference. Implemented in [9e13f89](https://github.com/engage-xr/RTLTMPro/pull/4/commits/9e13f896ba65ec6a957ef31d28e0ad77499cda2e)

**Review by ruben-ivre — APPROVED** [review]

File: Assets/RTLTMPro/Scripts/Runtime/TextUtils.cs (hunk 2 above)
Comment (ruben-ivre) [reply Assets/RTLTMPro/Scripts/Runtime/TextUtils.cs]: It's less risky using numerical values, as you can accidentally break file encoding thus breaking those chars, but you cannot accidentally break numerical values.

Comment (matthew-ivre) [reply Assets/RTLTMPro/Scripts/Runtime/TextUtils.cs]: Sorry, yeah, I meant I presumed there was no difference in this situation between (char)0x00AB and '\u00AB'.

Comment (ruben-ivre) [reply Assets/RTLTMPro/Scripts/Runtime/TextUtils.cs]: No, there's no diff, using numeric constants casted to char or directly using char literals makes no diff. https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/char#literals

**Review by daniel-ivre — APPROVED** [review]

**Review by juanma-ivre — APPROVED** [review]

**Review by BenIVRE — APPROVED** [review]

**Pull request merged**